### PR TITLE
[`Tests`/`LoRA`/`PEFT`] Test also on PEFT / transformers / accelerate latest

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Adds a simple logic in the workflow for the current PEFT + diffusers CI to make sure everything works on latest versions and make sure future PRs on diffusers about LoRA are backward compatible with latest peft / accelerate and transformers versions.

### CODE REVIEW

File: .github/workflows/pr_test_peft_backend.yml [hunk 1]
```diff
@@ -50,16 +45,19 @@ jobs:
       run: |
         apt-get update && apt-get install libsndfile1-dev libgl1 -y
         python -m pip install -e .[quality,test]
-        python -m pip install git+https://github.com/huggingface/accelerate.git
-        python -m pip install -U git+https://github.com/huggingface/transformers.git
-        python -m pip install -U git+https://github.com/huggingface/peft.git
+        if [ "${{ matrix.lib-version }}" == "main" ]; then
+            python -m pip install -U git+https://github.com/huggingface/peft.git
+            python -m pip install -U git+https://github.com/huggingface/transformers.git
+            python -m pip install -U git+https://github.com/huggingface/transformers.git
```
Comment (sayakpaul) [inline .github/workflows/pr_test_peft_backend.yml:51]: Duplicate.

File: .github/workflows/pr_test_peft_backend.yml [hunk 2]
```diff
@@ -50,16 +45,19 @@ jobs:
...
+        else
+            python -m pip install -U peft transformers accelerate
```
Comment (sayakpaul) [inline .github/workflows/pr_test_peft_backend.yml:53]: Having a hard time understanding the thought process behind this. 


I'd consider this level of version discrepancy to be generally a red flag. Could you elaborate here a bit?

File: .github/workflows/pr_test_peft_backend.yml (hunk 1 above)
Comment (younesbelkada) [inline .github/workflows/pr_test_peft_backend.yml:51]:
```suggestion
            python -m pip install -U git+https://github.com/huggingface/accelerate.git
```

File: .github/workflows/pr_test_peft_backend.yml (hunk 2 above)
Comment (younesbelkada) [reply .github/workflows/pr_test_peft_backend.yml:53]: this is just to make sure PRs we do in diffusers will not break anything for users that have the latest version of peft and transformers from pypi so it ensures backward compatbility. Right now we need to check these things manually which is a bit painful

**Pull request merged**